pkg/api/2018-09-30-preview/default.go

```diff
@@ -17,6 +17,27 @@ func setDefaults(oc *OpenShiftManagedCluster) {
 		oc.Properties.MasterPoolProfile.Count = to.Int64Ptr(3)
 	}
 
+	if len(oc.Properties.AgentPoolProfiles) == 0 {
+		oc.Properties.AgentPoolProfiles = []AgentPoolProfile{
+			{
+				Name: to.StringPtr("infra"),
+				Role: (*AgentPoolProfileRole)(to.StringPtr("infra")),
+			},
+		}
+	}
+
+	for i := range oc.Properties.AgentPoolProfiles {
```

Comment:
nit: I think we need to start putting some small comments on these cases :) Note for myself more than a comment to change :)